src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/ConsiderPassingBaseTypesAsParametersTests.cs

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.ConsiderPassingBaseTypesAsParameters,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.UnitTests
+{
+    public class ConsiderPassingBaseTypesAsParametersTests
```

Comment:
Potential additional tests:

- Method is marked as `override`
- Method is interface implementation
- Visibility user option
- Skip analysis user option
- Generic interfaces
- Struct
- Record

All the VB.NET tests

Reply:
@Evangelink Consider generic method test as well. I'm concerned by this comment in the compiler:

https://github.com/dotnet/roslyn/blob/10f51778f79b90cb0b7cdd39355584cef7fa7059/src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs#L480-L481

You might need `.OriginalDefinition` after `.TargetMethod` to fix it.

Reply:
Can also consider adding `.OriginalDefinition` inside `GetOverriddenMember` extension itself.